### ISSUE TITLE
OG-USA: Open for alternative gov't budget closure

### ISSUE THREAD

**Issue by jdebacker** [OpenSourcePolicyCenter/webapp-public#482]
On OG-USA landing page, we should add:
- Check box/radio button to toggle between budget rules: (i) transfers adjusted so that budget balances each period and (ii) No balanced budget, gov't spending adjusted after 20 years to hit SS debt/gdp ratio.
- Input box for target steady state debt/gdp ratio.

We'll have hover-overs for info that describes each of these rules.

Question:
- Do we need to display a note on OG-USA output page when gov't spending needs to go negative to close the budget?

CC @rickecon @salimfurth

**Comment by salimfurth** [comment]
I do think we should alert users when their options have pushed G negative.

**Comment by jdebacker** [comment]
Ok - we'll  need to figure out how to display that in the table.  

What could be really neat is to produce a graph showing the relevant fiscal rations over the transition path: taxes/gdp, debt/gdp, gov't spending gdp (maybe transfers/gdp, but that is assumed constant).

**Comment by salimfurth** [comment]
Agreed - either a graph or table of the ratios. Worth showing transfers even though it's fixed; it reminds users where the money is going.

**Comment by PeterDSteinberg** [comment]
@jdebacker Can you clarify `OG-USA landing page`: does that mean the landing page after you press `Link to Dynamic Simulation` button, e.g. this URL - [https://www.ospc.org/dynamic/10539/?start_year=2017](https://www.ospc.org/dynamic/10539/?start_year=2017) ?

**Comment by jdebacker** [comment]
@PeterDSteinberg No, I mean the page where one enters the parameters for the OG-USA model.  It is what you see when you were able to click the "Overlapping Generations Simulation" button on the page you shared.  I think it's [this](https://github.com/OpenSourcePolicyCenter/webapp-public/blob/master/templates/dynamic/dynamic_input_form.html) page.

**Comment by PeterDSteinberg** [comment]
Hi @jdebacker @rickecon .  I am working on this and issue #481 with @brittainhard 's help.  Can you clarify a couple things for me:

* Regarding the following items, can you explain how these parameters should be passed to the OG-USA `execute.py` functions - are OG-USA modifications needed or are these parameters already supported/named?
> Check box/radio button to toggle between budget rules: (i) transfers adjusted so that budget balances each period and (ii) No balanced budget, gov't spending adjusted after 20 years to hit SS debt/gdp ratio.
> Input box for target steady state debt/gdp ratio.
* I've been out of OG-USA for a couple months until last week or two.  Can you explain how the code should quickly check for this criterion below so the output page can be modified:
> when gov't spending needs to go negative to close the budget?

**Comment by jdebacker** [comment]
@PeterDSteinberg asks:
> Regarding the following items, can you explain how these parameters should be passed to the OG-USA execute.py functions - are OG-USA modifications needed or are these parameters already supported/named?

* A boolean can be passed as a kwarg to the `execute.runner()` functions that switches on/off the unbalanced budget.  If `budget_balance=False`, then the budget closure rule is used.
* The other parameter of the unbalanced budget model is the SS debt to GDP ratio.  This parameter is named `debt_ratio_ss` and can be passed to `execute.runner()`  in the `user_params` dictionary kwarg.


Also

> I've been out of OG-USA for a couple months until last week or two. Can you explain how the code should quickly check for this criterion below so the output page can be modified:
when gov't spending needs to go negative to close the budget?

We don't have this coded up yet.  @rickecon do you want this initially?  I'm not sure how it would display on the web app...  If we do add, I would do something like, in `SS.py` at around line 1030, add:
```
if output['Gss'] < 0:
   print('Warning: The combination of the tax policy you specified and your target debt-to-GDP ratio results in an infeasible amount of government spending in order to close the budget (i.e., G < 0)'
```




cc @rickecon

**Comment by PeterDSteinberg** [comment]
@jdebacker (apologies for missing this for a 6 days) - My thought is we should complete the work as well as we can now to take and pass the right inputs, perhaps deferring the check discussed above for `gov't spending needs to go negative to close the budget` to a separate issue.  We can go ahead and add the snippet to OG-USA (I'll make that PR now):

```
if output['Gss'] < 0:
   print('Warning: The combination of the tax policy you specified and your target debt-to-GDP ratio results in an infeasible amount of government spending in order to close the budget (i.e., G < 0)'
```
We may make a separate issue to ensure that the warning above is conveyed to the webapp - not sure exactly what is involved there yet if we want to put that warning on the results page.

cc @brittainhard

**Comment by PeterDSteinberg** [comment]
PR in OG-USA with the `Gss` warning: https://github.com/open-source-economics/OG-USA/pull/319
